sdk/openai/Azure.AI.OpenAI/src/Custom/AzureOpenAIExtensions.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System.ClientModel;
+using System.ClientModel.Primitives;
+using Azure.Core;
+
+namespace Azure.AI.OpenAI;
+
+/// <summary>
+/// The Azure OpenAI extensions.
+/// </summary>
+public static class AzureOpenAIExtensions
+{
+    /// <summary>
+    /// Gets the OpenAI chat client.
+    /// </summary>
+    /// <param name="provider"></param>
+    /// <param name="deploymentName"></param>
+    /// <returns></returns>
+    public static ChatClient GetAzureOpenAIChatClient(this ConnectionProvider provider, string? deploymentName = null)
+    {
+        string name = deploymentName ?? "default";
+
+        ChatClient chatClient = provider.Subclients.GetClient(() =>
+        {
+            AzureOpenAIClient aoiaClient = provider.Subclients.GetClient(() => CreateAzureOpenAIClient(provider), null);
+            return provider.CreateChatClient(aoiaClient, deploymentName);
+        }, name);
+
+        return chatClient;
+    }
+
+    /// <summary>
+    /// Gets the OpenAI embeddings client.
+    /// </summary>
+    /// <param name="provider"></param>
+    /// <param name="deploymentName"></param>
+    /// <returns></returns>
+    public static EmbeddingClient GetAzureOpenAIEmbeddingsClient(this ConnectionProvider provider, string? deploymentName = null)
```

Comment:
Should this be `GetAzureOpenAIEmbeddingClient` (i.e., "Embedding" instead of "Embeddings") in order to match the name of the `EmbeddingClient` class?